src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/CodeDirectiveFormattingTest.cs

```diff
@@ -940,13 +945,121 @@ public async Task CodeBlock_ObjectCollectionArrayInitializers(bool useSourceText
             Bar = new string[] {
                 ""Hello"",
                 ""There""
+            },
+            Baz = new string[]
+            {
+                ""Hello"",
+                ""There""
             }
         }
     };
 }
 ");
         }
 
+        [Theory]
+        [CombinatorialData]
+        [WorkItem("https://github.com/dotnet/razor-tooling/issues/6092")]
+        public async Task CodeBlock_ArrayInitializers(bool useSourceTextDiffer)
+        {
+            // The C# Formatter doesn't touch these types of initializers, so nor do we. This test
+            // just verifies we don't regress things and start moving code around.
+            await RunFormattingTestAsync(useSourceTextDiffer: useSourceTextDiffer,
+input: @"
+@code {
+    private void M()
+    {
+        var entries = new string[]
+        {
+            ""a"",
+            ""b"",
+            ""c""
+        };
+    }
+}
+",
+expected: @"@code {
+    private void M()
+    {
+        var entries = new string[]
+        {
+            ""a"",
+            ""b"",
+            ""c""
+        };
+    }
+}
+");
+        }
+
+        [Theory]
+        [CombinatorialData]
+        [WorkItem("https://github.com/dotnet/razor-tooling/issues/6092")]
+        public async Task CodeBlock_ObjectInitializers(bool useSourceTextDiffer)
```

Comment:
Added this test, and the below, just to make sure nothing else is wrong, or breaks. When I originally did array initializers formatting I intermingled all of the initializer formats together, which hid this bug.